pkg/storage/vault/aws/aws.go

```diff
@@ -0,0 +1,135 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/scratchdata/scratchdata/config"
+)
+
+type AWSVault struct {
+	client *secretsmanager.Client
+	prefix string
+}
+
+func NewAWSVault(conf map[string]any, destinations []config.Destination) (*AWSVault, error) {
```

Comment:
Do not pass `[]config.Destination` into this function. That should only be passed to the in-memory implementation.